acoustid/data/fingerprint.py

```
@@ -246,12 +247,17 @@ def _search_directly(
+        # Use the original or_ function but with proper typing
+        combined_condition = sql.or_(*conditions)
+
         query = self._create_search_query(
-            length, sql.or_(*conditions), max_results=max_results, compare_to=fp
+            length, combined_condition, max_results=max_results, compare_to=fp
         )
         if self.timeout:
             timeout_ms = int(self.timeout * 1000)
-            self.db.execute("SET LOCAL statement_timeout TO {}".format(timeout_ms))
+            self.db.execute(
+                text("SET LOCAL statement_timeout TO {}".format(timeout_ms))
```

Ruff (0.8.2)

Use f-string instead of `format` call

Convert to f-string

(UP032)